pkg/argocd/argocd.go

```diff
@@ -675,17 +675,34 @@ func getApplicationSourceType(app *v1alpha1.Application) v1alpha1.ApplicationSou
 	return app.Status.SourceType
 }
 
-// getApplicationSource returns the main source of a Helm or Kustomize type of the application
+// getApplicationSource returns the source matching git-repository annotation
+// or the Helm/Kustomize source, or the primary source if no match is found
 func getApplicationSource(app *v1alpha1.Application) *v1alpha1.ApplicationSource {
+	// Get the git repository from annotations for write-back
+	gitRepo := app.Annotations["argocd-image-updater.argoproj.io/git-repository"]
+	logCtx := log.WithContext().AddField("application", app.GetName())
```

Comment:
when getting annotation values, we should follow the pattern used in other places, such as:
```
// Check whether application has our annotation set
		annotations := app.GetAnnotations()
		if _, ok := annotations[common.ImageUpdaterAnnotation]; !ok {

```